tests/test_events.py

```diff
@@ -275,3 +276,21 @@ def test_order_started(events_object, client, base_ddo_url):
     published_ddo = get_ddo(client, base_ddo_url, did)
     # TODO: currently the graph for v4 is WIP, need to replace this
     assert published_ddo["stats"]["consumes"] == -1
+
+
+def test_metadata_state_update(client, base_ddo_url, events_object):
+
+    web3 = events_object._web3  # get_web3()
+    block = web3.eth.block_number
+    _ddo = new_ddo(test_account1, web3, f"dt.{block}")
+    did = _ddo.id
+
+    send_create_update_tx("create", _ddo, bytes([2]), test_account1)
+    events_object.process_current_blocks()
+    published_ddo = get_ddo(client, base_ddo_url, did)
+    assert published_ddo["id"] == did
+
+    send_set_metadata_state_tx(_ddo, test_account1)
```

Comment:
please parameterize the send_set_metadata_state_tx to take in the state itself.
assert that after we send 2, the asset is deleted (there shouldn't be a published ddo)
assert that after we send 0, the asset is back again (there should be a published ddo with that id)